### ISSUE TITLE
No example to demonstrate DBusSignalSubscription.

### ISSUE THREAD

**Issue by yash1200** [canonical/dbus.dart#105]
I am working on a plugin which fetches battery percentage and status from linux platform. It's dbus interface definition is : 

```
<node name="/org/freedesktop/UPower/devices/DisplayDevice">
   <interface name="org.freedesktop.UPower.Device">
      <property name="Percentage" type="d" access="read" />
      <signal name="State" type="n" />
   </interface>
</node>
```

And generated dart file is : 
```
import 'package:dbus/dbus.dart';

class OrgFreedesktopUPowerDevicesDisplayDevice extends DBusRemoteObject {
  OrgFreedesktopUPowerDevicesDisplayDevice(
      DBusClient client, String destination,
      {DBusObjectPath path = const DBusObjectPath.unchecked(
          '/org/freedesktop/UPower/devices/DisplayDevice')})
      : super(client, destination, path);

  /// Gets org.freedesktop.UPower.Device.Percentage
  Future<double> get percentage async {
    var value =
        await getProperty('org.freedesktop.UPower.Device', 'Percentage');
    return (value as DBusDouble).value;
  }

  /// Subscribes to org.freedesktop.UPower.Device.State
  Future<DBusSignalSubscription> subscribeState(
      void Function() callback) async {
    return await subscribeSignal('org.freedesktop.UPower.Device', 'State',
        (values) {
      if (values.length != 0) {
        callback();
      }
    });
  }
}

```
I want to return a Stream from plugin which states the status of battery in linux platform but can't find any documentation to implement it or a method to get it's value from generated `subscribeState` function.

**Comment by robert-ancell** [comment]
The current code uses callbacks, but it in process of being replaces with streams - https://github.com/canonical/dbus.dart/issues/81

**Comment by robert-ancell** [comment]
The stream changes have landed in 0.0.0-dev.20 and there is an example in example/signals.dart. Please re-open if you have any more issues!

**Comment by yash1200** [comment]
Thanks @robert-ancell .
